In [2]:
import pandas as pd
wholefile = pd.read_csv('//Documents//Test_sets//Files_for_Disc//testerselectionfiles//All_sound.csv')
ss = pd.read_csv('//Documents//Test_sets//Files_for_Disc//testerselectionfiles//Selection.csv')
wholefile.head()


FileNotFoundError: File b'//Documents//Test_sets//Files_for_Disc//testerselectionfiles//All_sound.csv' does not exist

In [2]:
import numpy as np

In [3]:
import matplotlib as p

In [14]:
wholefile.describe()

X            Y  SCAMIN  SCAMAX  EXPSOU  NOBJNM  OBJNAM  \
count  1066.000000  1066.000000     0.0     0.0     0.0     0.0     0.0   
mean     -4.149720    50.352354     NaN     NaN     NaN     NaN     NaN   
std       0.012985     0.008298     NaN     NaN     NaN     NaN     NaN   
min      -4.175090    50.333334     NaN     NaN     NaN     NaN     NaN   
25%      -4.160273    50.346614     NaN     NaN     NaN     NaN     NaN   
50%      -4.149808    50.354418     NaN     NaN     NaN     NaN     NaN   
75%      -4.138527    50.358979     NaN     NaN     NaN     NaN     NaN   
max      -4.124366    50.365973     NaN     NaN     NaN     NaN     NaN   

       QUASOU  SOUACC  STATUS     ...       INFORM  NINFOM  RECDAT  RECIND  \
count  1066.0     0.0     0.0     ...          0.0     0.0     0.0     0.0   
mean      1.0     NaN     NaN     ...          NaN     NaN     NaN     NaN   
std       0.0     NaN     NaN     ...          NaN     NaN     NaN     NaN   
min       1.0     NaN     NaN     ...          NaN     NaN     NaN     NaN   
25%       1.0     NaN     NaN     ...          NaN     NaN     NaN     NaN   
50%       1.0     NaN     NaN     ...          NaN     NaN     NaN     NaN   
75%       1.0     NaN     NaN     ...          NaN     NaN     NaN     NaN   
max       1.0     NaN     NaN     ...          NaN     NaN     NaN     NaN   

       SORDAT  iprown  iprtxt  $orenc  $pcano        Depth  
count     0.0     0.0     0.0     0.0     0.0  1066.000000  
mean      NaN     NaN     NaN     NaN     NaN     9.760483  
std       NaN     NaN     NaN     NaN     NaN     6.949285  
min       NaN     NaN     NaN     NaN     NaN    -3.000000  
25%       NaN     NaN     NaN     NaN     NaN     5.687750  
50%       NaN     NaN     NaN     NaN     NaN     7.423500  
75%       NaN     NaN     NaN     NaN     NaN    10.531250  
max       NaN     NaN     NaN     NaN     NaN    38.986000  

[8 rows x 24 columns]

In [13]:
ss.describe()

X           Y  SCAMIN  SCAMAX  EXPSOU  NOBJNM  OBJNAM  QUASOU  \
count  121.000000  121.000000     0.0     0.0     0.0     0.0     0.0   121.0   
mean    -4.144160   50.358472     NaN     NaN     NaN     NaN     NaN     1.0   
std      0.003022    0.001823     NaN     NaN     NaN     NaN     NaN     0.0   
min     -4.149466   50.355111     NaN     NaN     NaN     NaN     NaN     1.0   
25%     -4.146896   50.357023     NaN     NaN     NaN     NaN     NaN     1.0   
50%     -4.143884   50.358551     NaN     NaN     NaN     NaN     NaN     1.0   
75%     -4.141680   50.360011     NaN     NaN     NaN     NaN     NaN     1.0   
max     -4.139071   50.361506     NaN     NaN     NaN     NaN     NaN     1.0   

       SOUACC  STATUS     ...      NINFOM  RECDAT  RECIND  SORDAT  SORIND  \
count     0.0     0.0     ...         0.0     0.0     0.0     0.0     0.0   
mean      NaN     NaN     ...         NaN     NaN     NaN     NaN     NaN   
std       NaN     NaN     ...         NaN     NaN     NaN     NaN     NaN   
min       NaN     NaN     ...         NaN     NaN     NaN     NaN     NaN   
25%       NaN     NaN     ...         NaN     NaN     NaN     NaN     NaN   
50%       NaN     NaN     ...         NaN     NaN     NaN     NaN     NaN   
75%       NaN     NaN     ...         NaN     NaN     NaN     NaN     NaN   
max       NaN     NaN     ...         NaN     NaN     NaN     NaN     NaN   

       iprown  iprtxt  $orenc  $pcano       Depth  
count     0.0     0.0     0.0     0.0  121.000000  
mean      NaN     NaN     NaN     NaN   11.866512  
std       NaN     NaN     NaN     NaN    7.299043  
min       NaN     NaN     NaN     NaN    2.904000  
25%       NaN     NaN     NaN     NaN    6.494000  
50%       NaN     NaN     NaN     NaN    9.271000  
75%       NaN     NaN     NaN     NaN   16.338000  
max       NaN     NaN     NaN     NaN   33.233000  

[8 rows x 25 columns]

#Know from above that we have 1066 soundings and a selection of 121

#Need to remove columns which are not useful

In [5]:
wholefile_d= wholefile.loc[:,['X','Y','Depth','fid']]
wholefile_d.head(3)

X          Y   Depth                  fid
0 -4.153100  50.342776   7.711  GB 0044236486 00001
1 -4.162966  50.339084  10.000  GB 0044236488 00001
2 -4.163225  50.341281   5.751  GB 0044236489 00001

#old method

#wholefile_c= wholefile.dropna(axis = 'columns',how='any')

#wholefile_c.describe()

#dropped all columns with NaN values, know from describe above that all soundings are QUASOU 1 so this can be removed too

#wholefile_d = wholefile_c.drop('QUASOU', axis =1)


#need to add a column to say if the sounding has been selected 

In [6]:
#ss2 = ss.dropna(axis = 'columns',how='any')
ss2 = ss.loc[:,['X','Y','Depth']]

In [7]:
ss2['Selection'] = 1
ss2.columns

Index(['X', 'Y', 'Depth', 'Selection'], dtype='object')

In [8]:
wholefile_e = pd.merge(wholefile_d,ss2,how = 'outer',on =['X','Y'])
wholefile_e.head(10)
# checked depth values the same wholefile_e[wholefile_e.Selection == 1]
# Now need to make unselected Selection = 0 and and delete Depth_y

X          Y  Depth_x                  fid  Depth_y  Selection
0 -4.153100  50.342776    7.711  GB 0044236486 00001      NaN        NaN
1 -4.162966  50.339084   10.000  GB 0044236488 00001      NaN        NaN
2 -4.163225  50.341281    5.751  GB 0044236489 00001      NaN        NaN
3 -4.166210  50.340903    5.200  GB 0044236490 00001      NaN        NaN
4 -4.167520  50.340732    5.589  GB 0044236491 00001      NaN        NaN
5 -4.153763  50.341100   10.466  GB 0044236492 00001      NaN        NaN
6 -4.165542  50.340207    4.969  GB 0044236493 00001      NaN        NaN
7 -4.166643  50.340196    5.361  GB 0044236494 00001      NaN        NaN
8 -4.163679  50.340719    5.470  GB 0044236495 00001      NaN        NaN
9 -4.160701  50.340282    9.870  GB 0044236496 00001      NaN        NaN

In [12]:
wholefile_f = wholefile_e.drop('Depth_y', axis =1)
wholefile_f.describe()

X            Y      Depth_x  Selection
count  1066.000000  1066.000000  1066.000000      121.0
mean     -4.149720    50.352354     9.760483        1.0
std       0.012985     0.008298     6.949285        0.0
min      -4.175090    50.333334    -3.000000        1.0
25%      -4.160273    50.346614     5.687750        1.0
50%      -4.149808    50.354418     7.423500        1.0
75%      -4.138527    50.358979    10.531250        1.0
max      -4.124366    50.365973    38.986000        1.0

In [10]:
#replace NaN

final = wholefile_f.fillna(value = 0)
final.head(5)

X          Y  Depth_x                  fid  Selection
0 -4.153100  50.342776    7.711  GB 0044236486 00001        0.0
1 -4.162966  50.339084   10.000  GB 0044236488 00001        0.0
2 -4.163225  50.341281    5.751  GB 0044236489 00001        0.0
3 -4.166210  50.340903    5.200  GB 0044236490 00001        0.0
4 -4.167520  50.340732    5.589  GB 0044236491 00001        0.0

final.describe()

geopandas - buffer to a tollerance
density of data?
